In [ ]:
#pip install selenium

In [1]:
import selenium

In [2]:
import pickle
import requests
from bs4 import BeautifulSoup as BS

In [3]:
with open("./kbo.pkl", "rb") as f:
    kbo = pickle.load(f)

In [4]:
len(kbo)

829

In [5]:
kbo_player_url = "https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId={}"

In [21]:
#cnt = 0
for idx, id_ in enumerate(kbo):
    if idx % 100 == 0:
    #cnt += 1
    #print (cnt, end= ", ")
        print(idx, kbo_player_url.format(id_))

0 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=69158
100 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=50848
200 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=62920
300 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=66160
400 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=51546
500 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=52641
600 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=50040
700 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=77532
800 https://www.koreabaseball.com/Record/Player/HitterDetail/Basic.aspx?playerId=51702
64760


In [7]:
r = requests.get(kbo_player_url.format(id_))

In [44]:
bs = BS(r.text, 'html.parser')
bs


<!DOCTYPE html>

<html lang="ko">
<head><title>
	선수(타자) | 기록실 | KBO
</title><meta content="text/html; charset=utf-8" http-equiv="Content-Type"/><meta content="text/javascript" http-equiv="Content-Script-Type"/><meta content="text/css" http-equiv="Content-Style-Type"/><meta content="IE=edge, Chrome" http-equiv="X-UA-Compatible"/><meta content="KBO" name="title"/><meta content="스포츠투아이" name="author"/><meta content="KBO, 한국야구위원회, 프로야구, KBO 리그, 퓨처스리그, 프로야구순위, 프로야구 일정" name="keywords"/><meta content="width=1200" name="viewport"/><meta content="website" property="og:type"/><meta content="KBO 홈페이지" property="og:title"/><meta content="KBO, 한국야구위원회, 프로야구, KBO 리그, 퓨처스리그, 프로야구순위, 프로야구 일정" property="og:description"/><meta content="//lgcxydabfbch3774324.cdn.ntruss.com/KBO_IMAGE/KBOHome/resources/images/common/open_graph_kbo.png" property="og:image"/><meta content="//www.koreabaseball.com/" property="og:url"/>
<link href="//lgcxydabfbch3774324.cdn.ntruss.com/KBO_IMAGE/KBOHome/resources/images/commo

## 선수 각각은 dict 저장, dict -> list 담는다.
[ {선수1}, {선수1}.... {선수829} ]

### 혼자서 해보기

In [9]:
textp = BS(str(bs.find("div", class_="player_basic").find("ul"))).text
textp

'\n선수명: 황영국\n등번호: No.58\n생년월일: 1995년 12월 26일\n포지션: 투수(좌투좌타)\n신장/체중: 185cm/83kg\n경력: 청주우암초-청주중-청주고-한화-경찰\n입단 계약금: 20000만원\n연봉: 3400만원\n지명순위: 14 한화 1차\n입단년도: 14한화\n'

In [10]:
bs.find("form", action="./Basic.aspx?playerId={}".format(id_))

<form action="./Basic.aspx?playerId=64760" id="mainForm" method="post">
<input id="__VIEWSTATE" name="__VIEWSTATE" type="hidden" value="/wEPDwUKMTc0Mjc4MzYwNQ9kFgJmD2QWAmYPZBYCZg9kFgICAQ9kFgJmD2QWAgIBD2QWAgIDD2QWEGYPZBYYZg8WAh4EVGV4dAW/ATxoNCBpZD0iaDRUZWFtImNsYXNzPSJ0ZWFtIHJlZ3VsYXIvMjAyMi9lbWJsZW1fSEgiPjxzcGFuIGNsYXNzPSdlbWInPjxpbWcgc3JjPScvL2xnY3h5ZGFiZmJjaDM3NzQzMjQuY2RuLm50cnVzcy5jb20vS0JPX0lNQUdFL2VtYmxlbS9yZWd1bGFyLzIwMjIvZW1ibGVtX0hILnBuZycgLz48L3NwYW4+7ZWc7ZmUIOydtOq4gOyKpDwvaDQ+ZAICDw8WBB4ISW1hZ2VVcmwFSy8vbGdjeHlkYWJmYmNoMzc3NDMyNC5jZG4ubnRydXNzLmNvbS9LQk9fSU1BR0UvcGVyc29uL21pZGRsZS8yMDIyLzY0NzYwLmpwZx4NQWx0ZXJuYXRlVGV4dAUJ7Zmp7JiB6rWtFgIeB29uZXJyb3IFZHRoaXMuc3JjPScvL2xnY3h5ZGFiZmJjaDM3NzQzMjQuY2RuLm50cnVzcy5jb20vS0JPX0lNQUdFL0tCT0hvbWUvcmVzb3VyY2VzL2ltYWdlcy9jb21tb24vbm8taW1nLmdpZidkAgQPDxYCHwAFCe2ZqeyYgeq1rWRkAgYPDxYCHwAFAjU4ZGQCCA8PFgIfAAUTMTk5NeuFhCAxMuyblCAyNuydvGRkAgoPDxYCHwAFFO2IrOyImCjsooztiKzsooztg4ApZGQCDA8PFgIfAAUKMTg1Y20vODNrZ2RkAg4PDxYCHwAFMeyyreyjvOyasOyVlOy0iC3s

In [11]:
textp2 = bs.find("form", action="./Basic.aspx?playerId={}".format(id_)).find("div", class_="player_basic").find("ul").text
playert2 = textp2.strip().split('\n')
playert2

['선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화']

In [12]:
textp.split('\n')

['',
 '선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화',
 '']

In [13]:
playert = textp.strip().split('\n')
playert

['선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화']

In [14]:
asdf = []
for id_ in kbo:
    for id_ in playert2:
        asdf.append(id_)
print(asdf)

['선수명: 황영국', '등번호: No.58', '생년월일: 1995년 12월 26일', '포지션: 투수(좌투좌타)', '신장/체중: 185cm/83kg', '경력: 청주우암초-청주중-청주고-한화-경찰', '입단 계약금: 20000만원', '연봉: 3400만원', '지명순위: 14 한화 1차', '입단년도: 14한화', '선수명: 황영국', '등번호: No.58', '생년월일: 1995년 12월 26일', '포지션: 투수(좌투좌타)', '신장/체중: 185cm/83kg', '경력: 청주우암초-청주중-청주고-한화-경찰', '입단 계약금: 20000만원', '연봉: 3400만원', '지명순위: 14 한화 1차', '입단년도: 14한화', '선수명: 황영국', '등번호: No.58', '생년월일: 1995년 12월 26일', '포지션: 투수(좌투좌타)', '신장/체중: 185cm/83kg', '경력: 청주우암초-청주중-청주고-한화-경찰', '입단 계약금: 20000만원', '연봉: 3400만원', '지명순위: 14 한화 1차', '입단년도: 14한화', '선수명: 황영국', '등번호: No.58', '생년월일: 1995년 12월 26일', '포지션: 투수(좌투좌타)', '신장/체중: 185cm/83kg', '경력: 청주우암초-청주중-청주고-한화-경찰', '입단 계약금: 20000만원', '연봉: 3400만원', '지명순위: 14 한화 1차', '입단년도: 14한화', '선수명: 황영국', '등번호: No.58', '생년월일: 1995년 12월 26일', '포지션: 투수(좌투좌타)', '신장/체중: 185cm/83kg', '경력: 청주우암초-청주중-청주고-한화-경찰', '입단 계약금: 20000만원', '연봉: 3400만원', '지명순위: 14 한화 1차', '입단년도: 14한화', '선수명: 황영국', '등번호: No.58', '생년월일: 1995년 12월 26일', '포지션: 투수(좌투좌타)', '신장/체중: 185cm/83kg', '경력: 청주우암초-청주중-청주

In [15]:
len(asdf)

8290

In [23]:
text3 = BS(str(BS(r.text).find("div", class_="player_basic").find("ul"))).text
text3

'\n선수명: 황영국\n등번호: No.58\n생년월일: 1995년 12월 26일\n포지션: 투수(좌투좌타)\n신장/체중: 185cm/83kg\n경력: 청주우암초-청주중-청주고-한화-경찰\n입단 계약금: 20000만원\n연봉: 3400만원\n지명순위: 14 한화 1차\n입단년도: 14한화\n'

In [17]:
playert3 = text3.strip().split('\n')
playert3

['선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화']

In [19]:
asdf3 = []
for id_ in kbo:
    for x in playert3:
        asdf3.append(x)
asdf3

['선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화',
 '선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화',
 '선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화',
 '선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화',
 '선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포지션: 투수(좌투좌타)',
 '신장/체중: 185cm/83kg',
 '경력: 청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금: 20000만원',
 '연봉: 3400만원',
 '지명순위: 14 한화 1차',
 '입단년도: 14한화',
 '선수명: 황영국',
 '등번호: No.58',
 '생년월일: 1995년 12월 26일',
 '포

## 강사님이 하신

In [43]:
player_dict = dict()
for x in bs.find("div", class_ = "player_basic").findAll("li"):
    a, b = x.text.split(":")
    player_dict[a] = b.strip()
player_dict

{'선수명': '황영국',
 '등번호': 'No.58',
 '생년월일': '1995년 12월 26일',
 '포지션': '투수(좌투좌타)',
 '신장/체중': '185cm/83kg',
 '경력': '청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금': '20000만원',
 '연봉': '3400만원',
 '지명순위': '14 한화 1차',
 '입단년도': '14한화'}

In [93]:
cnt = 0
player_all = list()
for idx, id_ in enumerate(kbo):
    if idx % 30 == 0:
        print(idx)
    r = requests.get(kbo_player_url.format(id_))
    player_dict = dict()
    bs = BS(r.text, 'html.parser')
    for x in bs.find("div", class_ = "player_basic").findAll("li"):
        a, b = x.text.split(":")
        player_dict[a] = b.strip()
    player_all.append(player_dict)

0
30
60
90
120
150
180
210
240
270
300
330
360
390
420
450
480
510
540
570
600
630
660
690
720
750
780
810


In [53]:
player_all[0]

{'선수명': '강정현',
 '등번호': 'No.100',
 '생년월일': '1995년 09월 23일',
 '포지션': '투수(우투우타)',
 '신장/체중': '180cm/90kg',
 '경력': '남부민초-대신중-부경고-원광대-LG-상무',
 '입단 계약금': '7000만원',
 '연봉': '3300만원',
 '지명순위': '19 LG 2차 4라운드 35순위',
 '입단년도': '19LG'}

In [140]:
a, b

('90kg', ' 14한화')

## KBO 최고 연봉은? (금액)
* 주의사항 : 외국인 선수는 환율 1200원 계산

In [85]:
player_dict = dict()
for x in bs.find("div", class_ = "player_basic").find('span', id="cphContents_cphContents_cphContents_playerProfile_lblSalary").text:


3
4
0
0
만
원


In [89]:
cnt = 0
player_alla = list()
for idx, id_ in enumerate(kbo):
    if idx % 30 == 0:
        print(idx)
    r = requests.get(kbo_player_url.format(id_))
    player_dict = dict()
    bs = BS(r.text, 'html.parser')
    for x in bs.find("div", class_ = "player_basic").find('span', id="cphContents_cphContents_cphContents_playerProfile_lblSalary").text:
        player_alla.append(player_dict)
    player_alla     

0
30
60
90
120
150
180
210
240
270
300
330
360
390
420
450
480
510
540
570
600
630
660
690
720
750
780
810


[{},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},
 {},


In [91]:
bs.find("div", class_ = "player_basic").find('span', id="cphContents_cphContents_cphContents_playerProfile_lblSalary")

<span id="cphContents_cphContents_cphContents_playerProfile_lblSalary">3400만원</span>

## 강사님 코드

In [97]:
total = []
for x in player_all:
#     print(x['연봉'])
    if x['연봉'] == '':
        continue
    연봉 = int(x['연봉'][:-2])
    if x['연봉'][-2:] == "달러":
        연봉 = 연봉 * 1200
    else:
        연봉 = 연봉 * 10000
    total.append(연봉)   

In [98]:
total.sort()

In [99]:
total[-1]

8100000000

In [100]:
for x in player_all:
#     print(x['연봉'])
    if x['연봉'] == '':
        continue
    연봉 = int(x['연봉'][:-2])
    if x['연봉'][-2:] == "달러":
        연봉 = 연봉 * 1200
    else:
        연봉 = 연봉 * 10000
    x['연봉'] = 연봉

In [101]:
maximum = 0
a = dict()
for x in player_all:
    if x['연봉'] != "":
        if x['연봉'] > maximum:
            maximum = x['연봉']
            a['max'] = x

In [102]:
for x in player_all:
    x
    break

In [103]:
x['신장/체중']

'180cm/90kg'

In [104]:
tmp = []
for a in x['신장/체중'].split("/"):
    tmp.append(int(a.replace("cm", "").replace("kg","")))

In [105]:
tmp

[180, 90]

In [106]:
height, weight = [int(a.replace("cm", "").replace("kg","")) for a in x['신장/체중'].split("/")]

In [107]:
bmi = weight / (height / 100) ** 2

In [108]:
bmi

27.777777777777775

In [109]:
for x in player_all:
    height, weight = [int(a.replace("cm", "").replace("kg","")) for a in x['신장/체중'].split("/")]
    x['BMI'] = weight / (height / 100) ** 2

## BMI 지수가 가장 높은 선수?

In [110]:
maximum = 0
rt = dict()
for x in player_all:
    if maximum < x['BMI']:
        maximum = x['BMI']
        rt['maximum'] = [x['선수명'], x['BMI']]

In [111]:
rt

{'maximum': ['한기원', 39.24646781789639]}

In [112]:
x

{'선수명': '황영국',
 '등번호': 'No.58',
 '생년월일': '1995년 12월 26일',
 '포지션': '투수(좌투좌타)',
 '신장/체중': '185cm/83kg',
 '경력': '청주우암초-청주중-청주고-한화-경찰',
 '입단 계약금': '20000만원',
 '연봉': 34000000,
 '지명순위': '14 한화 1차',
 '입단년도': '14한화',
 'BMI': 24.251278305332356}

In [113]:
tmp = {}
for x in player_all:
    tmp[x['선수명']] = x['BMI']

In [114]:
sorted(tmp.items(), key=lambda x : x[1], reverse=True)

[('한기원', 39.24646781789639),
 ('이대호', 34.541396535232224),
 ('이성원', 33.30898466033601),
 ('수아레즈', 33.2409972299169),
 ('박상준', 32.82413836636788),
 ('최형우', 32.716049382716044),
 ('한동희', 32.60475787948315),
 ('김승현', 32.407407407407405),
 ('지시완', 32.05030371478282),
 ('김도규', 32.009548611111114),
 ('박진', 32.00096606690013),
 ('진승현', 31.899810964083173),
 ('이태오', 31.855955678670362),
 ('김주현', 31.795583304428252),
 ('나지완', 31.69907016060862),
 ('안승한', 31.63739669421488),
 ('황대인', 31.561671506122963),
 ('나균안', 31.50653254711527),
 ('박병호', 31.26369612856099),
 ('김태연', 31.246054791061734),
 ('조무근', 30.75),
 ('정보근', 30.693877551020407),
 ('안치홍', 30.614821360939274),
 ('푸이그', 30.556813037573566),
 ('서의태', 30.55585078116697),
 ('이승현', 30.457762250291136),
 ('서준원', 30.38714390065741),
 ('폰트', 30.336384869392468),
 ('놀린', 30.336384869392468),
 ('임석진', 30.24691358024691),
 ('주성원', 30.24691358024691),
 ('성동현', 30.234315948601665),
 ('노성호', 30.189590629151066),
 ('김기연', 29.983587930816814),
 ('추신수', 29

In [141]:
sorted(tmp)

['강경학',
 '강도훈',
 '강동연',
 '강동형',
 '강로한',
 '강매성',
 '강민국',
 '강민성',
 '강민호',
 '강백호',
 '강병우',
 '강산',
 '강상원',
 '강승호',
 '강원진',
 '강윤구',
 '강이준',
 '강재민',
 '강정현',
 '강진성',
 '강태경',
 '강태율',
 '강한울',
 '강현구',
 '강효종',
 '경우진',
 '고명성',
 '고명준',
 '고성민',
 '고승민',
 '고영창',
 '고영표',
 '고우석',
 '고종욱',
 '고효준',
 '공민규',
 '곽빈',
 '구본혁',
 '구승민',
 '구자욱',
 '구준범',
 '구창모',
 '권광민',
 '권동진',
 '권민석',
 '권성준',
 '권정웅',
 '권혁경',
 '권휘',
 '권희동',
 '김강률',
 '김강민',
 '김강현',
 '김건우',
 '김건태',
 '김겸재',
 '김광현',
 '김교람',
 '김규남',
 '김규성',
 '김규연',
 '김기연',
 '김기중',
 '김기탁',
 '김기환',
 '김녹원',
 '김대우',
 '김대유',
 '김대한',
 '김대현',
 '김도규',
 '김도영',
 '김도윤',
 '김도현',
 '김동엽',
 '김동우',
 '김동욱',
 '김동주',
 '김동준',
 '김동진',
 '김동혁',
 '김리안',
 '김만수',
 '김명신',
 '김문수',
 '김민기',
 '김민성',
 '김민수',
 '김민식',
 '김민우',
 '김민재',
 '김민혁',
 '김범수',
 '김병준',
 '김병희',
 '김상민',
 '김상수',
 '김서준',
 '김서진',
 '김석환',
 '김선기',
 '김선빈',
 '김선우',
 '김성균',
 '김성민',
 '김성우',
 '김성윤',
 '김성진',
 '김성표',
 '김성현',
 '김성협',
 '김성훈',
 '김세민',
 '김수윤',
 '김수환',
 '김승준',
 '김승현',
 '김시앙',
 '김시완',
 '김시훈',
 '김양수',
 '김영규',
 '김영웅',
 '김영준',
 '김영현',
 '김

- 파생변수

## 키 -> 나이
## KBO 선수의 평균 나이?
## 포지션별 등록 선수?

In [122]:
print(int(player_all[0]['생년월일'].split('년')[0]))
2022 - int(player_all[0]['생년월일'][:4]) + 1

1995


28

In [124]:
for x in player_all:
    x['나이'] = 2022 - int(x['생년월일'][:4]) + 1

In [125]:
player_all

[{'선수명': '강정현',
  '등번호': 'No.100',
  '생년월일': '1995년 09월 23일',
  '포지션': '투수(우투우타)',
  '신장/체중': '180cm/90kg',
  '경력': '남부민초-대신중-부경고-원광대-LG-상무',
  '입단 계약금': '7000만원',
  '연봉': 33000000,
  '지명순위': '19 LG 2차 4라운드 35순위',
  '입단년도': '19LG',
  'BMI': 27.777777777777775,
  '나이': 28},
 {'선수명': '강효종',
  '등번호': 'No.43',
  '생년월일': '2002년 10월 14일',
  '포지션': '투수(우투우타)',
  '신장/체중': '184cm/86kg',
  '경력': '저동초(일산서구리틀)-충암중-충암고',
  '입단 계약금': '20000만원',
  '연봉': 30000000,
  '지명순위': '21 LG 1차',
  '입단년도': '21LG',
  'BMI': 25.401701323251416,
  '나이': 21},
 {'선수명': '고우석',
  '등번호': 'No.19',
  '생년월일': '1998년 08월 06일',
  '포지션': '투수(우투우타)',
  '신장/체중': '182cm/90kg',
  '경력': '갈산초-양천중-충암고',
  '입단 계약금': '30000만원',
  '연봉': 270000000,
  '지명순위': '17 LG 1차',
  '입단년도': '17LG',
  'BMI': 27.17063156623596,
  '나이': 25},
 {'선수명': '구본혁',
  '등번호': 'No.4',
  '생년월일': '1997년 01월 11일',
  '포지션': '내야수(우투우타)',
  '신장/체중': '177cm/75kg',
  '경력': '중대초-잠신중-장충고-동국대',
  '입단 계약금': '5000만원',
  '연봉': 70000000,
  '지명순위': '19 LG 2차 6라운드 55순위',
  '입단년

In [130]:
sum([x['나이'] for x in player_all]) / len([x['나이'] for x in player_all])

26.700844390832327

In [133]:
print(max([x['나이'] for x in player_all]))
min([x['나이'] for x in player_all])

41


19

In [134]:
position = dict()
for x in player_all:
    if x['포지션'] in position:
        position[x['포지션']] += 1
    else:
        position[x['포지션']] = 1 # position 값을 초기화 라고 하심

In [135]:
sorted(position.items(), key=lambda x : x[1], reverse=True)

[('투수(우투우타)', 254),
 ('내야수(우투우타)', 118),
 ('투수(좌투좌타)', 106),
 ('포수(우투우타)', 66),
 ('내야수(우투좌타)', 60),
 ('외야수(우투우타)', 59),
 ('외야수(우투좌타)', 49),
 ('외야수(좌투좌타)', 35),
 ('투수(우언우타)', 30),
 ('투수(우투좌타)', 27),
 ('포수(우투좌타)', 9),
 ('내야수(좌투좌타)', 6),
 ('투수(좌투우타)', 3),
 ('투수(우투양타)', 3),
 ('투수(우언좌타)', 1),
 ('외야수(우투양타)', 1),
 ('내야수(우투양타)', 1),
 ('투수(우언양타)', 1)]

In [139]:
len(position.keys())

18